app/gameboard_api.py

```diff
@@ -73,3 +75,67 @@ def _find_original_pid(self, relationships, trait, value):
     @staticmethod
     def _get_sorted_links(ops):
         return sorted([lnk for op in ops for lnk in op.chain if lnk.finish and lnk.cleanup == 0], key=lambda i: i.finish)
+
+    def _get_points(self, red_op, blue_op, exchanges):
+        red_points = 0
+        blue_points = 0
+        for exchange in exchanges.values():
+            red_points, blue_points = self._calculate_red_links(blue_op, exchange, red_points, blue_points)
+            blue_points = self._calculate_blue_links(red_op, exchange, blue_points)
```

Comment:
Here I would not pass the blue_points arg it doesn't seem to be needed -- also I would rename to something about points clalculate_blue_point_etc
blue_points += self._calculate_blue_links(red_op, exchange)

Reply:
same for calc_red_links